packages/dev/core/src/Meshes/Compression/dracoEncoder.ts

```diff
@@ -0,0 +1,293 @@
+ * `AddFloatAttribute(mesh: Mesh, attribute: number, count: number, itemSize: number, array: TypedArray)`
+ * where `attribute = EncoderModule[<dracoAttribute>]`, `itemSize = <size>`, `array = <data>`, and count is the number of position vertices.
+ * @internal
+ */
+function PrepareAttributesForDraco(input: Mesh | Geometry, excludedAttributes?: string[]): Array<IDracoAttributeData> {
+    const attributes: Array<IDracoAttributeData> = [];
+
+    for (const kind of input.getVerticesDataKinds()) {
+        if (excludedAttributes?.includes(kind)) {
+            if (kind === VertexBuffer.PositionKind) {
+                throw new Error("Cannot exclude position attribute from Draco encoding.");
+            }
+            continue;
+        }
+
+        // Convert number[] to typed array, if needed
+        let data = input.getVerticesData(kind)!;
+        if (!(data instanceof Float32Array)) {
+            data = Float32Array.from(data!);
+        }
+        attributes.push({ babylonAttribute: kind, dracoAttribute: getDracoAttributeName(kind), size: input.getVertexBuffer(kind)!.getSize(), data: data });
+    }
+
+    return attributes;
+}
+
+/**
+ * Copies `data` of each `attribute` into a single buffer for fast transfer to the worker.
+ * @internal
+ */
+function UseTransferableBuffer(attributes: Array<IDracoAttributeData>, indices: Nullable<Uint32Array | Uint16Array>): ArrayBuffer {
+    // Allocate the buffer
+    let byteSize = 0;
+    for (const attribute of attributes) {
+        byteSize += attribute.data.byteLength;
+    }
+    byteSize += indices ? indices.byteLength : 0;
+
+    const buffer = new ArrayBuffer(byteSize);
+
+    // Copy attributes and indices data into buffer, remapping attributes.data to new bufferviews for the buffer
+    let offsetBytes = 0;
+    for (const attribute of attributes) {
+        const bufferView = new Float32Array(buffer, offsetBytes, attribute.data.length);
+        bufferView.set(attribute.data);
+        attribute.data = bufferView;
+        offsetBytes += bufferView.byteLength;
+    }
+    if (indices) {
+        const bufferView = new (indices instanceof Uint32Array ? Uint32Array : Uint16Array)(buffer, offsetBytes, indices.length);
+        bufferView.set(indices);
+    }
+
+    return buffer;
+}
+
+const DefaultEncoderOptions: IDracoEncoderOptions = {
+    decodeSpeed: 5,
+    encodeSpeed: 5,
+    method: "MESH_EDGEBREAKER_ENCODING",
+    quantizationBits: {
+        POSITION: 14,
+        NORMAL: 10,
+        COLOR: 8,
+        TEX_COORD: 12,
+        GENERIC: 12,
+    },
+};
+
+/**
+ * @experimental This class is subject to change.
+ *
+ * Draco Encoder (https://google.github.io/draco/)
+ *
+ * This class wraps the Draco encoder module.
+ *
+ * By default, the configuration points to a copy of the Draco encoder files from the Babylon.js preview cdn https://preview.babylonjs.com/draco_encoder_wasm_wrapper.js.

```

Comment:
We don't use `preview` anymore. It's only there for backwards compat.

```suggestion
 * By default, the configuration points to a copy of the Draco encoder files from the Babylon.js preview cdn https://cdn.babylonjs.com/draco_encoder_wasm_wrapper.js.
```

Reply:
as a future note - preview and cdn are both pointing towards the same place on the storage account. they always deliver the same file.